helm/cluster-vsphere/templates/provider-specific-files.yaml

```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  {{/*
+      You MUST bump the name suffix here and in `values.schema.json` every time one of these files
+      changes its content. Automatically appending a hash of the content here doesn't work
+      since we'd need to edit `values.schema.json` as well, but that file is created by humans.
+  */}}
+  name: {{ include "resource.default.name" $ }}-provider-specific-files-1
+  namespace: {{ $.Release.Namespace | quote }}
+data:
+  set-hostname.sh: {{ tpl ($.Files.Get "files/opt/bin/set-hostname.sh") $ | b64enc | quote }}
+type: Opaque
```

Comment:
the `cluster` chart currently doesn't have the ability to pass arbitrary file contents to the `files` section of ignition, so instead we create a secret which stores the contents of the the file(s) and then the secret's contents are written out to the file on disk (see https://github.com/giantswarm/cluster-vsphere/pull/263/files#diff-91218d0fa586a5a23b86c05ebd3ed9ed502aee6ace9953024119e735d7e1557dR157-R163)